Neo4j.Driver/Neo4j.Driver/Internal/Result/ResultReaderBuilder.cs

```diff
@@ -23,23 +23,20 @@
 
 namespace Neo4j.Driver.Internal.Result
 {
-    internal class ResultReaderBuilder : IMessageResponseCollector
+    internal class ResultReaderBuilder : BuilderBase
```

Comment:
Is it easy to break the logic of `IMessageResponseCollector` to an independent class such as `MessageResponseCollector` and have an field in this class?
This could also be done in another PR instead.